pkg/build/builder/common.go

```diff
@@ -84,15 +98,9 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 			},
 		},
 	}
-
+	// NOTE: not sure that this is needed anymore
 	// Reset ResourceVersion to avoid a conflict with other updates to the build
 	build.ResourceVersion = ""
```

Comment:
@bparees historically this is here because in the builder we had the build object that was passed with the build controller (via env var). That build object has already been modified by the time your build starts running. So we had 2 options:
1) Get the latest build object, but risk getting into a race with the build pod controller for the update
2) Update the revision regardless of the state of the rest of the build object.

This allows the latter. At the time the only function of the build/details endpoint was to allow you to update the revision.